src/libstd/sys/cloudabi/mutex.rs

```diff
@@ -74,14 +75,14 @@ impl ReentrantMutex {
             // If we fail to acquire the lock, it may be the case
             // that we've already acquired it and may need to recurse.
             if old & !abi::LOCK_KERNEL_MANAGED.0 == __pthread_thread_id.0 | abi::LOCK_WRLOCKED.0 {
-                *recursion += 1;
+                (*(*recursion).as_mut_ptr()) += 1;
```

Comment:
The outermost parentheses are not needed.

But also, are you sure this compiles? You need to get though both the `UnsafeCell` and the `MaybeUninit`. For windows this looks like `(&mut *self.inner.get()).as_mut_ptr()`.

Reply:
It does compile and pass all tests now (and now I'm sure that it is compiling the cloudabi lib). 

I was thinking that you deref the `UnsafeCell` then `as_mut_ptr` gives you a pointer to the type `MaybeUninit` wraps, which you then deref, so you get through both.

Reply:
You [cannot deref an `UnsafeCell`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=218f654f5104391a71e8b605bdf1e926), it's not a pointer.

Reply:
My bad, none of the `Cell` types implement `Deref`, should've looked more carefully.

It compiles because earlier in the method the local var `recursion` is bound to `self.recursion.get()`, so it holds a mut pointer not the`UnsafeCell`.